frontend/src/tree/TreeItemContextMenu.types.ts

```diff
@@ -24,6 +24,31 @@ export interface TreeItemContextMenuProps {
   selection: Selection;
   setSelection: (selection: Selection) => void;
   enterEditingMode: () => void;
-  deleteItem: () => void;
   onClose: () => void;
 }
+
+export interface GQLDeleteTreeItemData {
+  deleteTreeItem: GQLDeleteTreeItemPayload;
+}
+export interface GQLDeleteTreeItemPayload {
+  __typename: string;
+}
+
+export interface GQLDeleteTreeItemSuccessPayload extends GQLDeleteTreeItemPayload {
+  id: string;
+}
```

Comment:
Unless I missed something, you should have a GQLErrorPayload extending GQLDeleteTreeItemPayload with a message and you should use it